### PR TITLE
Generate generator jobs

### PR DESCRIPTION
**Pull request by M-Davies**
* Adds a generator job that, executed on git push to master, will regenerate the generator jobs
* Will also pass down params to the generator jobs to ensure they run correctly when it completes
* Will also trigger a building of said jobs after SUCCESSFULL completion to ensure everything is generated up to date
* Changes [regeneration/](https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/tree/master/pipelines/build/regeneration) directory name to `generation/`
* Moves all template files into their own directory and renames them to make more sense
* Doc and linter updates to account for changes

Closes: #10 
Signed-off-by: Morgan Davies <morgandavies2020@gmail.com>

### CODE REVIEW

**Comment by M-Davies** [comment]
Blocked by https://github.com/AdoptOpenJDK/ci-jenkins-pipelines/pull/42

File: docs/ARCHITECTURE.md [hunk 1]
```diff
@@ -14,9 +14,9 @@ A rough structure of the build is as follows
 
 ### Jenkins Job
 
-#### Regenerate Jobs
+#### Generate Jobs
 
-Jenkins CI jobs are generated by the [regeneration pipelines](https://ci.adoptopenjdk.net/job/build-scripts/job/utils/) in Jenkins. The documentation for this process can be found at [regeneration docs](https://github.com/AdoptOpenJDK/openjdk-build/tree/master/pipelines/build/regeneration), however they generate the jobs
+Jenkins CI jobs are generated by the [generation pipelines](https://ci.adoptopenjdk.net/job/build-scripts/job/utils/) in Jenkins. The documentation for this process can be found at [generation docs](https://github.com/AdoptOpenJDK/openjdk-build/tree/master/pipelines/build/generation), however they generate the jobs
```
Comment (gdams) [inline docs/ARCHITECTURE.md:19]:
```suggestion
Jenkins CI jobs are generated by the [generation pipelines](https://ci.adoptopenjdk.net/job/build-scripts/job/utils/) in Jenkins. The documentation for this process can be found at [generation docs](https://github.com/adoptium/ci-jenkins-pipelines/tree/master/pipelines/build/generation), however they generate the jobs
```

**Comment by karianna** [comment]
@M-Davies Would you be able to rebase this or should we close it?

**Comment by M-Davies** [comment]
Probably best to close. Last I checked, this isn't needed that badly and may just create more problems than it's worth

**Pull request closed without merging**